modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -2521,9 +2520,10 @@ private String exchangeTimingsLogMessage(String header, List<String> timings) {
             if (localReserved != null) {
                 boolean success = cctx.database().reserveHistoryForPreloading(localReserved);
 
-                // TODO: how to handle?
-                if (!success)
-                    err = new IgniteCheckedException("Could not reserve history");
+                if (!success) {
```

Comment:
In my understanding, the in-memory cluster will always print this warning message (please take a look at `IgniteCacheDatabaseSharedManager#reserveHistoryForPreloading()`). It does not make sense in this case, I think.

Reply:
I think that for in-memory cluster this check above will not be true, so we will not get here.

https://github.com/apache/ignite/blob/74b215580b4ebb7d883025259e7e29502b8e1a3e/modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java#L2518

Reply:
We discussed it personally.